WordPressAuthenticator/GoogleSignIn/URLRequest+GoogleSignIn.swift

```diff
@@ -0,0 +1,17 @@
+extension URLRequest {
+
+    static func googleSignInTokenRequest(
```

Comment:
Below, I wrote:

```
// The concern with extending a Foundation type with special-purpose domain object getting ability
// is that it would pollute the namespace, offering the protocol methods as an option everywhere
// `URLSession` is used, even in part of the app that are unrelated with the resource.
```

I guess one could raise the same concern for this method on `URLRequest` 🤔 

One option could be to use a builder object `GoogleSignInRequestBuilder` `tokenRequest(body:) throws -> URLRequest`.